File: server/internal/store/dict.go
package store

import "github.com/ubibot/ubibot-platform-open/internal/model"

func (s *Store) CreateDictEntry(typ, key, label string, sort int) (*model.DictEntry, error) {
	e := &model.DictEntry{Type: typ, Key: key, Label: label, Sort: sort}
	if err := s.db.Create(e).Error; err != nil {
		return nil, err
	}
	return e, nil
}

// ListDictEntries returns entries for typ (or every entry if typ is ""),
// ordered for direct use as a dropdown's option list.
func (s *Store) ListDictEntries(typ string) ([]model.DictEntry, error) {
	q := s.db.Order("sort asc, id asc")
	if typ != "" {
		q = q.Where("type = ?", typ)
	}
	var rows []model.DictEntry
	err := q.Find(&rows).Error
	return rows, err
}

func (s *Store) UpdateDictEntry(id uint, label string, sort int) error {
	return s.db.Model(&model.DictEntry{}).Where("id = ?", id).Updates(map[string]any{
		"label": label, "sort": sort,
	}).Error
}

func (s *Store) DeleteDictEntry(id uint) error {
	return s.db.Delete(&model.DictEntry{}, id).Error
}

File: admin/src/api/dict.ts
import { api } from './client'

export interface DictEntry {
  id: number
  type: string
  key: string
  label: string
  sort: number
}

export function listDictEntries(type?: string) {
  return api.get<{ list: DictEntry[] }>(`/api/admin/dict${type ? `?type=${encodeURIComponent(type)}` : ''}`)
}

export function createDictEntry(input: { type: string; key: string; label: string; sort: number }) {
  return api.post<DictEntry>('/api/admin/dict', input)
}

export function updateDictEntry(id: number, input: { label: string; sort: number }) {
  return api.patch<{ message: string }>(`/api/admin/dict/${id}`, input)
}

export function deleteDictEntry(id: number) {
  return api.del<{ message: string }>(`/api/admin/dict/${id}`)
}
